app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt

```diff
@@ -202,6 +202,8 @@ class SessionControlView(
                 override fun onLayoutCompleted(state: RecyclerView.State?) {
```

Comment:
Put the scrollToPosition here would mean that _every_ `view.layout` would scroll to the top, which may be undesirable. For example:

1. Scroll down to 'Recently visited'.
2. Click the 'Show all'.
3. Press the back button.

The expected result is that you are in the same scroll position as before, but now you are always taken to the top.

I haven't tried it yet, but maybe what you want is the same thing we do in the tabstray: perform a one-time scroll to position on layout and then remove the listener.

See: [`<VH : RecyclerView.ViewHolder> RecyclerView.Adapter<out VH>.observeFirstInsert`](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/tabstray/ext/RecyclerViewAdapter.kt)

Reply:
I tested it but it didn't scroll up.  It stay at the same scroll position.

Reply:
Hmm.. for me on Android 12 (Pixel 3) it jumps to the top in a flashing sort of way. Hmm.. it seems to do that when switching apps too.

Here's a video of what I see with your change (the first app is a debug build, the second is current nightly):

https://user-images.githubusercontent.com/1370580/147001598-aff225d5-c5b3-414f-9c7c-c99b5a5514d2.mp4

Reply:
That's weird.  I can't reproduce that.  I'll add the unregister oberver code to be safe though.  Not sure why i can't reproduce it with Android 10.